native/src/routes/NewsContainer.tsx

```diff
@@ -1,320 +1,78 @@
-      status: 'loading',
-      progress: 0
-    }
-  }
-
-  const cities = state.cities.models
-  const cityModel = cities.find(city => city.code === route.city)
 
   if (!cityModel) {
-    throw new Error('cityModel is undefined!')
+    return <LayoutContainer />
   }
 
-  if (route.status === 'loading') {
-    return {
-      status: 'success',
-      refreshProps,
-      innerProps: {
-        status: 'fetching',
-        newsId: route.newsId,
-        language: state.contentLanguage,
-        selectedNewsType: route.type,
-        cityModel,
-        navigation,
-        route: ownProps.route,
-        page: null,
-        news: null,
-        hasMoreNews: null,
-        isFetchingMore: null
-      }
-    }
-  }
-
-  return {
-    status: 'success',
-    refreshProps,
-    innerProps: {
-      status: 'ready',
-      newsId: route.newsId,
-      language: state.contentLanguage,
-      news: route.models,
-      selectedNewsType: route.type,
-      cityModel,
-      navigation,
-      route: ownProps.route,
-      hasMoreNews: route.status !== 'loadingMore' ? route.hasMoreNews : null,
-      page: route.status !== 'loadingMore' ? route.page : null,
-      isFetchingMore: route.status === 'loadingMore'
-    }
+  const navigateToNews = (newsType: NewsType) => {
+    setSelectedNewsId(null)
+    setSelectedNewsType(newsType)
   }
-}
-
-const NewsContainer = (props: ContainerPropsType) => {
-  const { cityModel, dispatch, selectedNewsType, route, language, newsId, navigation } = props
 
-  const fetchNews = useCallback(
-    (newsType: NewsType) => {
-      dispatch({
-        type: 'FETCH_NEWS',
-        params: {
-          city: cityModel.code,
-          language,
-          newsId: null,
-          type: newsType,
-          key: route.key,
-          criterion: {
-            forceUpdate: false,
-            shouldRefreshResources: false
-          }
-        }
-      })
-    },
-    [cityModel, language, route, dispatch]
-  )
-  const fetchMoreNews = useCallback(
-    ({
-      hasMoreNews,
-      news,
-      page
-    }: {
-      hasMoreNews: boolean | null | undefined
-      news: NewsModelsType
-      page: number | null
-    }) => () => {
-      if (!hasMoreNews || page === null) {
-        // Already fetching more
-        return
-      }
-
-      if (selectedNewsType === TU_NEWS_TYPE) {
-        const fetchNews: FetchMoreNewsActionType = {
-          type: 'FETCH_MORE_NEWS',
-          params: {
-            city: cityModel.code,
-            language,
-            newsId,
-            type: TU_NEWS_TYPE,
-            key: route.key,
-            page: page + 1,
-            previouslyFetchedNews: news,
-            hasMoreNews,
-            criterion: {
-              forceUpdate: false,
-              shouldRefreshResources: false
-            }
-          }
-        }
-        dispatch(fetchNews)
+  const changeUnavailableLanguage = (newLanguage: string) => {
+    setSelectedNewsId(null)
+    setSelectedLanguage(newLanguage)
```

Comment:
Will refactor this in a separate PR such that we don't have to keep a local state.